### PR TITLE
Fix android bug

### PR DESCRIPTION
**Pull request by peregrineshahin**
fix resize events keeps triggering old error alerts if the keyboard shows or hides on  mobile.

fixes: https://discord.com/channels/435943710472011776/900771437177094156/1160537676286541824

### CODE REVIEW

**Comment by vdbergh** [comment]
I think we still want the resize to happen no? Shouldn’t we only suppress the stale error message?

**Comment by vdbergh** [comment]
Hmmm this seems hard to achieve with the current code. The input validation should have been done on submission.

**Comment by vdbergh** [comment]
Perhaps the script should remember the last good parameters, and rewrite the input fields to the good parameters, if the new parameters don’t validate.

**Comment by peregrineshahin** [comment]
> I think we still want the resize to happen no? Shouldn’t we only suppress the stale error message?

Good point! Yes we should resize regardless

**Comment by peregrineshahin** [comment]
> Perhaps the script should remember the last good parameters, and rewrite the input fields to the good parameters, if the new parameters don’t validate.

100%

**Comment by peregrineshahin** [comment]
Well not really rewrite the input, Yes remember the old params to resize it again but not play with the input field

**Comment by vdbergh** [comment]
That's a matter of taste. I would feel uncomfortable that the graph shown may not correspond to the input fields. But I can also see why one would think otherwise. I have no strong feelings about this.

**Comment by vdbergh** [comment]
What about the URL arguments? If they are invalid then `draw_charts()` will be done with `sprt=null` it seems to me.

`valid_sprt` could be initialized with default values. But then the user would not know what they are since we are not updating the input fields.

**Comment by vdbergh** [comment]
Currently the graphs are blank if the URL arguments are invalid. I guess that makes sense too.

File: server/fishtest/static/js/calc.js [hunk 1]
```diff
@@ -60,35 +62,40 @@ function set_fields() {
...
-  if (val != "") {
-    alert(val);
-    return;
-  }
+  if (error) {
+    // do not show a stale alert with a resize
+    // and use last valid sprt to draw chart
+    if (!resize || !valid_sprt) {
+      alert(val);
+      return;
```
Comment (vdbergh) [inline server/fishtest/static/js/calc.js:95]: I guess you mean `alert(val)` here.

**Review by ppigazzini — APPROVED** [review]
Looks good on DEV and android

**Comment by ppigazzini** [comment]
PROD updated, thank you @peregrineshahin :)

**Pull request merged**